test/wait_for_blocks.go

```diff
@@ -0,0 +1,60 @@
+package test
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// ChainHeighter fetches the current chain block height.
+type ChainHeighter interface {
+	Height(ctx context.Context) (uint64, error)
+}
+
+// WaitForBlocks blocks until all chains reach a block height delta equal to or greater than the delta argument.
+func WaitForBlocks(parent context.Context, delta int, chains ...ChainHeighter) error {
+	if len(chains) == 0 {
+		panic("missing chains")
+	}
+	eg, ctx := errgroup.WithContext(parent)
```

Comment:
The naming we are using elsewhere is keeping `ctx` as the surrounding method's parameter, so this line looks like `eg, egCtx := errgroup.WithContext(ctx)`.